app/backend/src/main/java/com/wishlist/services/ShoppingListService.java

```diff
@@ -258,56 +258,21 @@ public ShoppingList updateList(String id, ShoppingList updatedShoppingList) thro
     }
 
     @Override
-    public ShoppingList bulkCheck(BulkCheckDTO dto, String listId) {
+    public ShoppingList bulkEdit(BulkEditDTO dto, String listId) {
 
         ShoppingList list = shoppingListRepository.findById(listId).orElseThrow(ShoppingListDoesNotExistException::new);
-        Map<String, Boolean> mapped = new HashMap<>();
-        Arrays.stream(dto.getIds()).forEach(id -> mapped.put(id, true));
-        List<ShoppingItem> updatedItems = new ArrayList<>();
-
-        if (!dto.isAllSelected()) {
-            list.getItemList().forEach(item -> {
-                if (mapped.containsKey(item.getId())) {
-                    item.setChecked(true);
-                }
-                updatedItems.add(item);
-            });
-        } else {
-            list.getItemList().forEach(item -> {
+        List<ShoppingItem> shoppingListItems = list.getItemList();
+        list.setItemList(dto.getItems());
+
+        if (dto.isAllSelected()) {
+            for(ShoppingItem item: shoppingListItems) {
                 item.setChecked(true);
-                updatedItems.add(item);
-            });
+            }
+            return list;
```

Comment:
If you're already doing bulkEdit, which receives a list of items and not ids, why not expand the method to possibly update the name also? Not always, but currentName!=received name -> setName would be good